fpdecimal.go

```diff
@@ -1,31 +1,52 @@
-package fp3
+package fpdecimal
 
 import (
-	"github.com/nikolaydubina/fpdecimal"
 	"github.com/nikolaydubina/fpdecimal/constraints"
 )
 
-// Decimal is fixed-point decimal with 3 fractions.
-// Actual value scaled by 1000x.
-// Values fit in ~9 quadrillion.
+// Decimal is a decimal with fixed number of fraction digits.
+// By default, uses 3 fractional digits.
+// For example, values with 3 fractional digits will fit in ~9 quadrillion.
 // Fractions lower than that are discarded in operations.
 // Max: +9223372036854775.807
 // Min: -9223372036854775.808
 type Decimal struct{ v int64 }
 
 var Zero = Decimal{}
 
-func FromInt[T constraints.Integer](v T) Decimal { return Decimal{int64(v) * 1000} }
+var multipliers = []int64{1, 10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000, 10000000000}
 
-func FromFloat[T constraints.Float](v T) Decimal { return Decimal{int64(v * 1000)} }
+var fractionDigits int8 = 3
+
+func SetFractionDigits(fraction uint8) bool {
+	if fraction < 0 || fraction > 10 {
+		return false
+	}
+	fractionDigits = int8(fraction)
+	return true
+}
```

Comment:
maybe good just to export pkg var as uint8. not sure really need this validation.
```suggestion
var FractionDigits unt8 = 3
```

Reply:
We need this check only because we have multipliers for 0..10 digits.
For another case, we need to inline valid **else block** (something like **return multipliers[FractionDigits] || 1**)